crates/bevy_reflect/src/func/mod.rs

```diff
@@ -0,0 +1,220 @@
+//! Reflection-based dynamic functions.
+//!
+//! This module provides a way to pass around and call functions dynamically
+//! using the [`Function`] type.
+//!
+//! Many simple functions and closures can be automatically converted to [`Function`]
+//! using the [`IntoFunction`] trait.
+//!
+//! Once the [`Function`] is created, it can be called with a set of arguments provided
+//! via an [`ArgList`].
+//!
+//! This returns a [`FunctionResult`] containing the [`Return`] value,
+//! which can be used to extract a [`Reflect`] trait object.
+//!
+//!
+//! # Example
+//!
+//! ```
+//! # use bevy_reflect::func::args::ArgList;
+//! # use bevy_reflect::func::{Function, FunctionResult, IntoFunction, Return};
+//! fn add(a: i32, b: i32) -> i32 {
+//!   a + b
+//! }
+//!
+//! let mut func: Function = add.into_function();
+//! let args: ArgList = ArgList::default().push_owned(25_i32).push_owned(75_i32);
```

Comment:
i think it would be nice to have the things pushed in the `ArgList` clearly be `dyn Reflect`, to show that the user doesn't need to have access their type

Reply:
Good point. I'll update the docs!